README.md

```diff
@@ -121,6 +121,35 @@ Just use `"test": "exodus-test"`
 
 - `--test-force-exit` -- force exit after tests are done (useful in integration tests where it could be unfeasible to resolve all open handles)
 
+## Jest compatibility
+
+The `--jest` mode is mostly compatible with Jest. There are some noteworthy differences though.
+This tool does not hoist mocks, so it is important that a mock is defined before the module that uses it is imported.
```

Comment:
While the reasoning here is correct for CJS, for ESM there is no such difference -- Jest also doesn't hoist mocks in ESM.

Ref: https://jestjs.io/docs/ecmascript-modules#module-mocking-in-esm

Now sure how to better describe this though

Reply:
I noted this here https://github.com/ExodusMovement/test/pull/19#issuecomment-2571727522 as well